pydantic/_internal/_config.py

```diff
@@ -257,6 +263,9 @@ def push(self, config_wrapper: ConfigWrapper | ConfigDict | None):
     json_schema_extra=None,
     strict=False,
     revalidate_instances='never',
+    val_date_or_time_unit='infer',
+    ser_date_or_time_unit='infer',
```

Comment:
Opinion: users probably want to have both of these the same in 99% of cases. I would prefer we come up with a design where the obvious thing is easy and succinct, and in the rare case where these need to be different users have a complex override. So it makes me sad to have two configs here.

... I know we have the same problem in `ser_json_bytes` and `val_json_bytes`, so maybe this is a V3 API refresh question.